en/services/parameter.md

```diff
@@ -108,23 +108,33 @@ sequenceDiagram;
+    Note over GCS: Timeout (drone sent all params)
+    GCS->>Drone: Request each missing param by index (PARAM_REQUEST_READ)
+    GCS-->>GCS: Start receive timeout
+    Drone->>GCS: Send param with PARAM_VALUE
 {% endmermaid %}
 
 The sequence of operations is:
 
-1. GCS (client) sends [PARAM_REQUEST_LIST](../messages/common.md#PARAM_REQUEST_READ) specifying the target system/component.
-1. Drone sends all parameters individually in [PARAM_VALUE](../messages/common.md#PARAM_VALUE) messages.
-   - The drone should allow a break between each message in order to avoid saturating the link.
-1. GCS accumulates parameters in order to know which parameters have been/not been received  (`PARAM_VALUE` contains total number of params and index of current param).
-1. GCS starts timeout after each `PARAM_VALUE` message in order to detect when parameters are no longer being sent.  
-1. After timeout (messages no longer being sent) the GCS can request any missing parameter values.
- 
-   The GCS determines what parameters are missing based on the `param_count` and `param_index` fields from received [PARAM_VALUE](../messages/common.md#PARAM_VALUE) messages.
-   The messages are [requested individually](#read_single) using [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ) and the missing `param_index`   
+1. GCS (client) sends [PARAM_REQUEST_LIST](../messages/common.md#PARAM_REQUEST_READ) specifying a target system/component.
+   - Broadcast addresses may be used. All targetted components should respond with parameters (or ignore the request if they have none).
+   - The GCS is expected to accumulate parameters from all responding systems.
+   - GCS starts a timeout timer and waits for parameters.
+     On timeout the GCS should fail gracefully (reset the state machine).
+1. The targetted component(s) should respond, sending all parameters individually in [PARAM_VALUE](../messages/common.md#PARAM_VALUE) messages.
+   - Allow breaks between each message in order to avoid saturating the link.
+   - Components with no parameters should ignore the request.
+1. GCS accumulates parameters for each compent in order to know which parameters have been/not been received (`PARAM_VALUE` contains total number of params and index of current param).
+1. GCS starts timeout after each `PARAM_VALUE` message in order to detect when parameters are no longer being sent.
+1. After timeout (All components finished sending params) the GCS individually requests any missing parameter values.
+   - The GCS determines what parameters are missing for each component based on the `param_count` and `param_index` fields from received [PARAM_VALUE](../messages/common.md#PARAM_VALUE) messages.
+   - The messages are [requested individually](#read_single) using [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ) and the missing `param_index`.
+   - A timeout is set on each request.
+     A GCS should fail gracefully on timeout.
```

Comment:
What do you think should happen if there is a timeout on the second phase where you are requesting the previously dropped params.
For example, say you have 10 components sending params and you get a timout (following retries) on one of them. I'd imagine the ideal being that you get all the other params for all the other components and report and error for any that aren't complete. Do you delete the params you did get and mark them incomplete in some way? Or should all that be GCS dependent (ie outside of spec)

Reply:
That is also GCS dependant as to whether it wants to run with partial params. QGC reports that it could get all params, and in that case it doesn't show vehicle setup pages since it doesn't really know which parameters are needed for that to work.